File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        Map<String, Integer> map = new HashMap<>();
        for(String whomParticipate : participant)
            map.put(whomParticipate, map.getOrDefault(whomParticipate, 0) + 1);
        for(String whomComplete : completion) {
            map.put(whomComplete, map.get(whomComplete) - 1);
            if(map.get(whomComplete) == 0)
                map.remove(whomComplete);
        }
        
        return map.keySet().iterator().next();
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    playerDict = dict()
    answer = ''

    for player in participant:
        if player in playerDict:
            playerDict[player] += 1
        else:
            playerDict[player] = 1

    for player in completion:
        playerDict[player] -= 1
        if playerDict.get(player) == 0:
            playerDict.pop(player)

    return playerDict.__iter__().__next__()